### PR TITLE
Fix #149 -  Fix `FieldError` on Nautobot v1.2.x when creating PR review

### LINKED ISSUE #149: Field Error when creating pull request review
Creating a pull request review using the form throws an exception:
```
<class 'django.core.exceptions.FieldError'>

Cannot resolve keyword 'user' into field. Choices are: id, pull_request, pull_request_id, reviewed_at, reviewer, reviewer_id, state, summary

Python version: 3.6.15
Nautobot version: 1.2.0b1
```

This is new behavior as this was working perfectly prior.

Here is an example of a pull request review create form:
image

When I click the "Create" button, I get this:
image

### PR DESCRIPTION
**Pull request by jathanism**
I narrowed down the issue to be related to this change in `nautobot.core.views.generic.ObjectEditView.get_object()`:

```diff
@@ -248,13 +283,23 @@ class ObjectEditView(GetReturnURLMixin, ObjectPermissionRequiredMixin, View):

     def get_object(self, kwargs):
         # Look up an existing object by slug or PK, if provided.
-        if "slug" in kwargs:
-            return get_object_or_404(self.queryset, slug=kwargs["slug"])
-        elif "pk" in kwargs:
-            return get_object_or_404(self.queryset, pk=kwargs["pk"])
-        # Otherwise, return a new instance.
+        if kwargs:
+            return get_object_or_404(self.queryset, **kwargs)
         return self.queryset.model()
```

If I revert the method to pre-1.2.0 it works. For now I dropped a `FIXME` comment in there to revisit this at a later date.

### CODE REVIEW

**Comment by tim-fiola** [comment]
I've tested the fix in this branch and I can now create pull request reviews

**Comment by tim-fiola** [comment]
image

File: dolt/views.py [hunk 1]
```diff
@@ -753,6 +744,17 @@ def alter_obj(self, obj, request, url_args, url_kwargs):  # noqa: D102
         obj.reviewer = request.user
         return obj
 
+    # TODO(jathan): Figure out why we had to revert to pre-1.2.0 `get_object()`
+    # for this to work.
```
Comment (glennmatthews) [inline dolt/views.py:748]: Presumably this is due to the `kwargs["user"] = request.user` at line 757 below.

**Review by glennmatthews — APPROVED** [review]
This looks fine as a workaround until we fix the issue in core. Should we open a tracking issue for reverting this workaround once that's fixed?

**Comment by jathanism** [comment]
I confirmed that https://github.com/nautobot/nautobot/pull/1145 fixes this and obviates the need for this PR.

**Pull request closed without merging**